js/components/user-bar.js

```diff
@@ -6,120 +6,146 @@ define([
 	'providers/Component',
 	'utils/CommonUtils',
 	'webapi/AuthAPI',
-	'less!./user-bar.less'
-], function (
-	ko,
+	'services/http',
+], function (ko,
 	view,
 	appConfig,
 	state,
 	Component,
 	commonUtils,
-	authApi
+	authApi,
+	httpService,
 ) {
 	class UserBar extends Component {
 		constructor(params) {
-			super(params);			
+			super(params);
+			var endpoint = appConfig.api.url + 'notifications';
 			this.model = params.model;
 			this.appConfig = appConfig;
 			this.token = authApi.token;
-			this.tokenExpired = authApi.tokenExpired;    
-			this.authLogin = authApi.subject;		
+			this.tokenExpired = authApi.tokenExpired;
+			this.authLogin = authApi.subject;
 			this.pollInterval = null;
 			this.isLoggedIn = ko.computed(() => {
 				return authApi.isAuthenticated();
 			});
 			this.loading = params.model.loading;
 
+			this.showJobModal = ko.observable(false);
+			this.jobListing = state.jobListing;
+
+			this.jobNotificationsPending = ko.computed(() => {
+				var unviewedNotificationCount = this.jobListing().filter(j => {
+					return !j.viewed();
+				}).length;
+				return unviewedNotificationCount;
+			});
+
+			this.clearJobNotificationsPending = this.clearJobNotificationsPending.bind(this);
+
+			this.updateJobStatus = this.updateJobStatus.bind(this);
 			this.startPolling = () => {

```

Comment:
Move methods out of constructor (https://github.com/OHDSI/Atlas/issues/990) to class-level ones